Sources/Foundation/AsyncLineSequence.swift

```diff
@@ -0,0 +1,166 @@
+    public typealias Element = String
+    
+    var base: Base
+    
+    @available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *)
+    public struct AsyncIterator: AsyncIteratorProtocol {
+        public typealias Element = String
+        
+        var byteSource: Base.AsyncIterator
+        var buffer: Array<UInt8> = []
+        var leftover: UInt8? = nil
+        
+        internal init(underlyingIterator: Base.AsyncIterator) {
+            byteSource = underlyingIterator
+        }
+        
+        // We'd like to reserve flexibility to improve the implementation of
+        // next() in the future, so aren't marking it @inlinable. Manually
+        // specializing for the common source types helps us get back some of
+        // the performance we're leaving on the table.
+        @_specialize(where Base == URL.AsyncBytes)
+        @_specialize(where Base == FileHandle.AsyncBytes)
+        //@_specialize(where Base == URLSession.AsyncBytes)
+        public mutating func next() async rethrows -> String? {
+            /*
+             0D 0A: CR-LF
+             0A | 0B | 0C | 0D: LF, VT, FF, CR
+             E2 80 A8:  U+2028 (LINE SEPARATOR)
+             E2 80 A9:  U+2029 (PARAGRAPH SEPARATOR)
+             */
+            let _CR: UInt8 = 0x0D
+            let _LF: UInt8 = 0x0A
+            let _NEL_PREFIX: UInt8 = 0xC2
+            let _NEL_SUFFIX: UInt8 = 0x85
+            let _SEPARATOR_PREFIX: UInt8 = 0xE2
+            let _SEPARATOR_CONTINUATION: UInt8 = 0x80
+            let _SEPARATOR_SUFFIX_LINE: UInt8 = 0xA8
+            let _SEPARATOR_SUFFIX_PARAGRAPH: UInt8 = 0xA9
+          
+            func yield() -> String? {
+                defer {
+                    buffer.removeAll(keepingCapacity: true)
+                }
+                if buffer.isEmpty {
+                    return nil
+                }
+                return String(decoding: buffer, as: UTF8.self)
+            }
+            
+            func nextByte() async throws -> UInt8? {
+                defer { leftover = nil }
+                if let leftover = leftover {
+                    return leftover
+                }
+                return try await byteSource.next()
+            }
+            
+            while let first = try await nextByte() {
+                switch first {
+                case _CR:
+                    let result = yield()
+                    // Swallow up any subsequent LF
+                    guard let next = try await byteSource.next() else {
+                        return result //if we ran out of bytes, the last byte was a CR
+                    }
+                    if next != _LF {
+                        leftover = next
+                    }
+                    if let result = result {
+                        return result
+                    }
+                    continue
+                case _LF..<_CR:
+                    guard let result = yield() else {
+                        continue
```

Comment:
Forgive me if I'm oversimplifying... there's a lot going on here. 😊  It looks like this completely removes empty lines, which seems suboptimal, given that...

1. blank lines are meaningful in countless data applications, and
2. it's easy enough to add your own `isEmpty` filter on the sequence if you really do want to strip them out.

Based on this, the default behavior should probably be to *include* empty lines (i.e., change `continue` to `return ""` in the `_CR` and `_LF` cases).

Reply:
If in doubt, please cross-check with Foundation's behavior. It needs to have the same semantics, else we widen the (already large) gap between APPLE and !APPLE.

Reply:
Ah, yeah, good point.  This does match Foundation's behavior, so the issue would be against Apple, not this code.  Thanks for the check!